T/TempestModel/build_tarballs.jl

```diff
@@ -0,0 +1,111 @@
+
+cp -v test/shallowwater_sphere/BarotropicInstabilityTest ${bindir}
+cp -v test/shallowwater_sphere/MountainRossbyTest ${bindir}
+cp -v test/shallowwater_sphere/RossbyHaurwitzWaveTest ${bindir}
+"""
+
+# Note: We are restricted to the platforms that NetCDF supports, the library is Unix only
+platforms = [
+    Platform("x86_64", "linux"),
+    Platform("x86_64", "macos"),
+] 
+platforms = expand_cxxstring_abis(platforms)
+  
+products = [
+    # non-hydro sphere test cases
+    ExecutableProduct("ScharMountainSphereTest", :ScharMountainSphereTest_exe),
+    ExecutableProduct("BaroclinicWaveJWTest", :BaroclinicWaveJWTest_exe),
+    ExecutableProduct("StationaryMountainFlowTest", :StationaryMountainFlowTest_exe),
+    ExecutableProduct("MountainWaveSphereTest", :MountainWaveSphereTest_exe),
+    ExecutableProduct("HeldSuarezTest", :HeldSuarezTest_exe),
+    ExecutableProduct("InertiaGravityWaveTest", :InertiaGravityWaveTest_exe),
+    ExecutableProduct("BaroclinicWaveUMJSTest", :BaroclinicWaveUMJSTest_exe),
+    ExecutableProduct("MountainRossby3DTest", :MountainRossby3DTest_exe),
+    ExecutableProduct("BaldaufGravityWaveTest", :BaldaufGravityWaveTest_exe),
+
+    # non-hydro xz test cases
+    ExecutableProduct("RobertBubbleCartesianTest", :RobertBubbleCartesianTest_exe),
+    ExecutableProduct("ShearJetMtnWave2DCartesianTest", :ShearJetMtnWave2DCartesianTest_exe),
+    ExecutableProduct("ThermalBubbleCartesian3DTest", :ThermalBubbleCartesian3DTest_exe),
+    ExecutableProduct("Baroclinic3DCartesianTest", :Baroclinic3DCartesianTest_exe),
+    ExecutableProduct("DensityCurrentCartesianTest", :DensityCurrentCartesianTest_exe),
+    ExecutableProduct("NonHydroMountainCartesianTest", :NonHydroMountainCartesianTest_exe),
+    ExecutableProduct("HydrostaticMountainCartesianTest", :HydrostaticMountainCartesianTest_exe),
+    ExecutableProduct("ThermalBubbleCartesianTest", :ThermalBubbleCartesianTest_exe),
+    ExecutableProduct("Baroclinic3DCartesianRidgeTest", :Baroclinic3DCartesianRidgeTest_exe),
+    ExecutableProduct("ScharMountainCartesianTest", :ScharMountainCartesianTest_exe),
+    ExecutableProduct("InertialGravityCartesianXZTest", :InertialGravityCartesianXZTest_exe),
+
+    # shallow water test cases
+    ExecutableProduct("BarotropicInstabilityTest", :BarotropicInstabilityTest_exe), 
+    ExecutableProduct("MountainRossbyTest", :MountainRossbyTest_exe),
+    ExecutableProduct("RossbyHaurwitzWaveTest", :RossbyHaurwitzWaveTest_exe),
+]
+
+dependencies = Dependency[
+    Dependency("MKL_jll"),
+    Dependency("MPICH_jll"),
+    Dependency("NetCDF_jll"),
+    Dependency("HDF5_jll"),
+    # The following is adapted from NetCDF_jll
+    Dependency("LibCURL_jll"),
+    # The following libraries are dependencies of LibCURL_jll which is now a
+    # stdlib, but the stdlib doesn't explicitly list its dependencies
+    Dependency("LibSSH2_jll"),
+    Dependency("MbedTLS_jll", v"2.24.0"),
+    Dependency("nghttp2_jll"),
```

Comment:
I'm still not convinced this is necessary.  We used to have a problem with stdlibs not being listed as dependencies, but that should have been fixed looong ago.  The only problem I'm aware of is that we get the wrong version of MbedTLS (no clue why, compat bounds in the registry look good, as far as I can tell) and that can be forced with
```suggestion
    # This is only an indirect dependency
    BuildDependency(PackageSpec(; name="MbedTLS_jll", version="2.24.0")),
```

Reply:
Ok i can try

Reply:
Looks like that worked :slightly_smiling_face:   I'm still puzzled by the fact we get the wrong version of MbedTLS, as far as I can tell LibCURL_jll specifies correctly the compatibility bound: https://github.com/JuliaRegistries/General/blob/7392039acb21210b43cef86c7fa5f6f42213952e/L/LibCURL_jll/Compat.toml#L7-L9

Reply:
Cool!  I'll update the other package where I used the same form as well.

Reply:
I believe I understood what's the problem!  Without doing anything special, Pkg installs `LibCURL_jll` 7.71, which doesn't have a compat with `MbedTLS_jll` and then installs the latest version of `MbedTLS_jll`, 2.28.  If I explicitly ask for LibCURL_jll 7.73 I get the right MbedTLS.  I think we should fix the compat with `MbedTLS_jll` also for previous versions of `LibCURL_jll`